ouroboros-network-framework/src/Simulation/Network/Snocket.hs

```diff
@@ -759,8 +759,7 @@ mkSnocket state tr = Snocket { getLocalAddr
                   conn <- mkConnection tr bearerInfo connId
                   writeTVar fdVarLocal (FDConnecting connId conn)
                   modifyTVar (nsConnections state)
-                             (Map.insert (normaliseId connId)
-                                         (dualConnection conn))
+                             (Map.insert (normaliseId connId) conn)
```

Comment:
I think this change is right but for a different reason that you presented last time.

We write `conn` to the queue elsewhere, and the inbound side will take it from the queue not from `nsConnections`.  That's why this has no impact on non simultanous TCP open connections.

But this will have an impact on the implementation of TCP sim open. That's where we relay on `nsConnections`.  When we read from `nsConnections` in a simultaneous tcp open case ([ref](https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network-framework/src/Simulation/Network/Snocket.hs#L831)), we apply `dualConnection`, that's why the one above is not needed.

Reply:
It would be good to add an explanation in the commit message.

Reply:
Not quite, when we write `conn` to the queue we write the one present in the nsConnections, see: https://github.com/input-output-hk/ouroboros-network/blob/b8691349731b4926a60d51df1c666b2f937eebc3/ouroboros-network-framework/src/Simulation/Network/Snocket.hs#L806

Yes I agree it needs a better commit message

Reply:
You're right :) and actually fixes both sim-open and normal-open.